server/src/main/java/org/apache/cassandra/sidecar/db/schema/SidecarInternalKeyspace.java

```diff
@@ -33,20 +33,19 @@
  * {@link org.apache.cassandra.sidecar.db.RestoreJob} table and {@link org.apache.cassandra.sidecar.db.RestoreSlice}
  * table
  */
-@Singleton
 public class SidecarInternalKeyspace extends AbstractSchema
 {
     private final SchemaKeyspaceConfiguration keyspaceConfig;
     private final boolean isEnabled;
-    private final Set<TableSchema> tableSchemas = new HashSet<>();
+    private final Set<TableSchema> tableSchemas = ConcurrentHashMap.newKeySet();
```

Comment:
Was this done just to get rid of the `synchronized` on registerTableSchema for stylistic reasons (as I can't imaging it was a significant performance issue), or was there a bug this fixed?

Assuming we really need to make this change, I wonder if we want to be specific about the type of tableSchemas to show it really needs to be concurrent:
```suggestion
    private final ConcurrentHashMap.KeySetView<TableSchema, Boolean> tableSchemas = ConcurrentHashMap.newKeySet();
```

It's a bit ugly just because of the type `newKeyset` returns, but it makes it a bit clearer that this needs to be a concurrent set (now, why the Java implementers didn't just give us a ConcurrentSet to make this clear, no idea).

Reply:
I think this is a class-internal detail, but I'd prefer to keep it as Set for now

Reply:
I think I wanted to avoid synchronization.